src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -3423,6 +3423,36 @@ static class D
             Diagnostic(ErrorCode.ERR_InterceptorSignatureMismatch, @"InterceptsLocation(""Program.cs"", 11, 9)").WithArguments("Program.InterceptableMethod(ref readonly int)", "D.Interceptor(in int)").WithLocation(17, 6));
     }
 
+    [Fact]
+    public void SignatureMismatch_10()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            struct Program
+            {
+                public void InterceptableMethod() => Console.Write("Original");
+
+                public static void Main()
+                {
+                    new Program().InterceptableMethod();
+                }
+            }
+
+            static class D
+            {
+                [InterceptsLocation("Program.cs", 10, 23)]
+                public static void Interceptor(this in Program x) => Console.Write("Intercepted");
+            }
+            """;
+        var comp = CreateCompilation(new[] { (source, "Program.cs"), s_attributesSource }, parseOptions: RegularWithInterceptors);
+        comp.VerifyEmitDiagnostics(
+            // Program.cs(16,6): error CS9148: Interceptor must have a 'this' parameter matching parameter 'ref Program this' on 'Program.InterceptableMethod()'.
+            //     [InterceptsLocation("Program.cs", 10, 23)]
+            Diagnostic(ErrorCode.ERR_InterceptorMustHaveMatchingThisParameter, @"InterceptsLocation(""Program.cs"", 10, 23)").WithArguments("ref Program this", "Program.InterceptableMethod()").WithLocation(16, 6));
+    }
```

Comment:
>} [](http://example.com/codeflow?start=4&length=1)

Consider testing `readonly void InterceptableMethod()` with `static void Interceptor(this in Program)`, `static void Interceptor(this ref readonly Program)`, and `static void Interceptor(this ref Program)`.